src/Compilers/Test/Core/Platform/CoreClr/CoreCLRRuntimeEnvironment.cs

```diff
@@ -104,11 +110,201 @@ public int Execute(string moduleName, string[] args, string expectedOutput)
         public SortedSet<string> GetMemberSignaturesFromMetadata(string fullyQualifiedTypeName, string memberName) =>
             GetEmitData().GetMemberSignaturesFromMetadata(fullyQualifiedTypeName, memberName);
 
+        private class Resolver : ILVerify.ResolverBase
```

Comment:
The `CoreCLRRuntimeEnvironment` class comes into play when we are executing our tests on .NET Core vs. NET Framework. The ILVerify toolchain though should be runnable on any IL. Or at worst should be runnable when compiling for .NET Core. What the test compiles for and what runtime the test executes on are different concepts.

That made me a bit surprised to see this type here. My intuition was that ILVerification would be a general verification pass we could run independent of which runtime our tests were executing on. 

Am I missing something here?

Reply:
PEVerify is in `DesktopRuntimeEnvironment` so I'd followed that model, but I think you're right: IL verification likely can be done in `CompilationVerifier` instead of `CoreCLRRuntimeEnvironment`/`DesktopRuntimeEnvironment`. It just needs the emitted bits and the bits for all the dependencies. I'll take a look at that.

We can discuss whether ILVerify should also be run on desktop. I had it that way in an earlier commit, but then ran into some hurdles (should be possible to resolve, but didn't want to block progress on main goal).